app/main.py

```diff
@@ -1,8 +1,31 @@
 class Person:
-    # write your code here
-    pass
+    people: dict = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    spouse_types: set[str] = {"wife", "husband"}
+    for person in people:
+        if person["name"] in Person.people:
+            continue
+        current_person = Person(person["name"], person["age"])
+        Person.people[person["name"]] = current_person
+        spouse_type: set = (spouse_types & person.keys())
+        if len(spouse_type) < 1 or person.get(next(iter(spouse_type))) is None:
+            continue
+        spouse_name: str = person.get(next(iter(spouse_type)))
+        if spouse_name not in Person.people:
+            spouse_info = next((p for p in people
+                                if p["name"] == spouse_name), None)
+            Person.people[spouse_name] = Person(spouse_info["name"],
+                                                spouse_info["age"])
```

Comment:
the same as before, why do u need this ?

Reply:
Agreed, this operation is redundant